### PR TITLE
Reducer documentation

### PR DESCRIPTION
**Pull request by VoxSciurorum**

### CODE REVIEW

**Comment by VoxSciurorum** [comment]
> * The 5th paragraph contains an explicit `&nbsp;` that should be removed.
> * To syntax-highlight the code consistently with the rest of the website, the code blocks should be surrounded by `` ```cpp `` and `` ``` `` (or by  `` ```c `` and `` ``` ``).

I wrote the documentation in emacs using standard markdown and it was correct as written.  Something in the uploading process quoted the &.  I will try to edit the file on my branch to fix it.

As for the code blocks, I don't think language tags are part of standard markdown.  I will add them if we are only displaying the code on this web page.

**Comment by behoppe** [comment]
Thanks @VoxSciurorum for documenting our new reducer functionality and tapping some reviewers. To complement this reference perspective, I wonder if we want to develop an OpenCilk tutorial that gets new parallel programmers interested in using reducers. I have collected Cilk Plus material along these lines into #146. It's very much a work in progress, but I'm curious if you agree that a tutorial along those lines would help?

File: src/doc/reference/reducers.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+program can give incorrect results, crash, or worse.  A counter may
+not increment reliably or a linked list may become corrupt.
+
+A reducer is a special case of a more general type known as a
+_hyperobject_.  Different types of hyperobjects are used depending on
+the desired semantics.
+
+Reducers are used when the final value in a variable is built up from
+a series of independent modifications.  The modifications should all
+be of the same kind, such as by adding a number to an accumulator or
```
Comment (wsmoses) [inline src/doc/reference/reducers.md:21]: ``Of the same kind'' is unclear

File: src/doc/reference/reducers.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
+---
+title: Reducers
+author: John F. Carr
+---
+
+# Reducers
```
Comment (neboat) [inline src/doc/reference/reducers.md:6]: I'm not sure how everything will look once this page is merged with style fixes in other PRs.  But right now I think this line can be removed, since the title "Reducers" already show up elsewhere on the page.

File: src/doc/reference/reducers.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+For example, to declare a reducer holding sums of `double`s
+one can write
+
+```c
+    void zero_double(void *view) { *(double *)view = 0.0; }
+    void add_double(void *left, void *right)
+        { *(double *)left += *(double *)right; }
+    double cilk_reducer(zero_double, add_double) sum;
+```
```
Comment (neboat) [inline src/doc/reference/reducers.md:77]: My inclination is to remove the leading indentation on the code within the code block (for this code block and all other code blocks in this document).  Right now this indentation appears to add unnecessary whitespace to the code on the deployed page.

**Review by neboat — COMMENTED** [review]
Added a couple of comments.  These comments are minor nits; I think this doc is essentially ready to deploy.
